skopt/searchcv.py

```diff
@@ -722,10 +616,16 @@ def fit(self, X, y=None, groups=None):
             # do the optimization for particular search space
             while n_iter > 0:
                 # when n_iter < n_jobs points left for evaluation
-                n_jobs_adjusted = min(n_iter, self.n_jobs)
+                n_jobs_adjusted = min(n_iter, n_jobs)
```

Comment:
we need to add something that protects us from doing too many calls towards the end: when `n_iter > 0 and n_iter < n_jobs_adjusted`. Say we have `n_iter = 3` and `n_jobs = 4` we would evaluate one model too many currently.